tjpcov/main.py

```diff
@@ -370,25 +385,174 @@ def get_tracer_info(self, two_point_data={}):
+            tracer_comb 2 (list): List of the pair of tracer names of C_ell^2
+            ccl_tracers ():
+            tracer_Noise ():
+            coupled (bool): True to return the coupled Gaussian covariance
+            (default False)
+            cache (dict): Dictionary with the necessary workspaces and
+            covariance workspaces. It accept masks (keys: 'm1', 'm2', 'm3',
+            'm4'), fields (keys: 'f1', 'f2', 'f3', 'f4'), workspaces (keys:
+            'w13', 'w23', 'w14', 'w24', 'w12', 'w34'), the covariance
+            workspace (key: 'cw') and a NmtBin (key: 'bins').
+
+        Returns:
+        --------
+            cov (dict):  Gaussian covariance matrix for a pair of C_ell. keys
+            are 'final' and 'final_b'. The covariance stored is the same in
+            both cases.
+
+        """
+        if coupled:
+            raise ValueError('Computing coupled covariance matrix not ' +
+                             'implemented yet')
+
+        if 'bins' in cache:
+            bins = cache['bins']
+            ell = np.arange(bins.lmax + 1)
+            ell_eff = bins.get_effective_ells()
+        else:
+            raise ValueError('Not yet implemented: you need to pass a'
+                             + 'NmtBin instance')
+            bins = self.bins
+            ell = np.arange(self.ell.max())
+            ell_eff = self.ell_bins
+
+        if 'cosmo' in cache:
+            cosmo = cache['cosmo']
+        else:
+            cosmo = self.cosmo
+
+        tr = {}
+        tr[1], tr[2] = tracer_comb1
+        tr[3], tr[4] = tracer_comb2
+
+        dof = {}
+        dof[12] = nmt_tools.get_tracer_comb_dof(self.cl_data, tracer_comb1)
+        dof[34] = nmt_tools.get_tracer_comb_dof(self.cl_data, tracer_comb2)
+        dof[13] = nmt_tools.get_tracer_comb_dof(self.cl_data, (tr[1], tr[3]))
+        dof[24] = nmt_tools.get_tracer_comb_dof(self.cl_data, (tr[2], tr[4]))
+        dof[14] = nmt_tools.get_tracer_comb_dof(self.cl_data, (tr[1], tr[4]))
+        dof[23] = nmt_tools.get_tracer_comb_dof(self.cl_data, (tr[2], tr[3]))
+
+        s = {}
+        s[1], s[2] = nmt_tools.get_tracer_comb_spin(self.cl_data, tracer_comb1)
+        s[3], s[4] = nmt_tools.get_tracer_comb_spin(self.cl_data, tracer_comb2)
+
+
+        # Fiducial cl
+        cl = {}
+        # Coupled noise
+        SN = {}
+        for i in [13, 24, 14, 23]:
+            # Fiducial cl
+            i1, i2 = [int(j) for j in str(i)]
+            key = f'cl{i}'
+            if key in cache:
+                cl[i] = cache[key]
+            else:
+                cl[i] = np.zeros((dof[i], ell.size))
+                cl[i][0] = ccl.angular_cl(cosmo, ccl_tracers[tr[i1]],
+                                          ccl_tracers[tr[i2]], ell)
+
+            # Coupled noise
+            key = f'SN{i}'
+            if key in cache:
+                SN[i] = cache[key]
+            else:
+                SN[i] = np.zeros((dof[i], ell.size))
```

Comment:
This can simply be SN[i] = np.zeros( ell.size), since SN is same for both E & B modes

Reply:
Yes, but NaMaster works with Cell arrays of shape (# dof, ell.size), so it's easier to have it in that format. Otherwise doing `signal + noise` would be a pain.

Reply:
If it is required by Namaster design, I am ok with having this format.